GomokuClient/packages/gomoku-story/src/components/TimeControls/TimeControls.tsx

```diff
@@ -12,45 +11,60 @@ export interface GameType {
 
 export interface TimeControlsProps {
   gameTypes: GameType[];
-  onCreateGame: (
-    boardSize: number,
-    timeControl?: SwaggerTypes.TimeControlDto,
-  ) => void;
+  onGameTypeSelected: (gameType: GameType) => void;
+  onGameTypeUnselected: () => void;
   isLoading: boolean;
 }
 
 export const TimeControls = ({
   gameTypes,
-  onCreateGame,
-  isLoading,
-}: TimeControlsProps) => (
-  <div className="grid grid-cols-2 gap-4 sm:grid-cols-3 sm:gap-6">
-    {gameTypes.map((game, index) => (
-      <GameCreatorButton
-        key={index}
-        timeControl={game.timeControl}
-        onCreateGame={onCreateGame}
-        isLoading={isLoading}
-      >
+  onGameTypeSelected,
+  onGameTypeUnselected,
+}: TimeControlsProps) => {
+  const [selectedGameType, setSelectedGameType] = useState<GameType>();
+
+  useEffect(() => {
+    return () => onGameTypeUnselected();
+  }, []);
```

Comment:
Could you please explain the purpose of this return statement in useEffect: return () => onGameTypeUnselected();? I believe this can be avoided when navigating to a specific route, as cleanup operations should ideally be triggered during route navigation

Reply:
Yeah, probably I can do clean-up in home page. Will remove this

Reply:
Removed and added cleanup in home page